test_util/integration_test.py

```diff
@@ -1340,3 +1349,148 @@ def wait_for_endpoint():
     r_data['Properties']['clusterId'] = ''
 
     assert r_data == exp_data
+
+
+class AgentManipulator:
+
+    def __init__(self, cluster, ssh_user, key_path):
+        self._cluster = cluster
+        self._ssh_user = cluster.ssh_user
+        self._ssh_key_path = cluster.ssh_key_path
+
+    def _multi_run(self, cmds):
+        with contextlib.closing(ssh_tunnel.SSHTunnel(
+            self._cluster.ssh_user,
+            self._cluster._ssh_key_path,
+            self._cluster.slaves[0])) as ssh_runner:
+            for cmd in cmds:
+                yield ssh_runner.remote_cmd(self.list_or_make(cmd))
+
+    def _run(self, cmd):
+        return ssh_tunnel.run_ssh_cmd(
+            self._cluster.ssh_user,
+            self._cluster._ssh_key_path,
+            self._cluster.slaves[0],
+            self.list_or_make(cmd)
+        )
+
+    @staticmethod
+    def list_or_make(v, split_fn=shlex.split):
+        return v if isinstance(v, list) else split_fn(v)
+
+    def _stop_mesos_agent(self):
+        self._run('systemctl stop dcos-mesos-slave')
+
+    def _start_mesos_agent(self, clear_state, restart=True):
+        if clear_state:
+            self._clear_agent_state()
+        self._run('systemctl start dcos-mesos-slave')
+
+    def _clear_agent_state(self):
+        # TODO: check this path
+        self._run('rm -rf /var/lib/mesos/slave')
+
+    def _run_volume_discovery(self, delete_only=False):
+        steps = (
+            'rm /var/lib/dcos/mesos-resources',
+            'systemctl start dcos-vol-discovery-priv-agent'
+        )
+        return self._multi_run(steps[:1] if delete_only else steps)
```

Comment:
Probably better to split _run_volume_discovery into two simple methods:
* `_start_volume_discovery(self)` to match the start_mesos_agent naming convention
* `_clear_volume_discovery_state(self)` to match the clear_agent_state naming convention